brave/browser/brave_browser_context.cc

```diff
@@ -181,25 +181,24 @@ BraveBrowserContext::BraveBrowserContext(
   }
 
   std::string tor_path;
-  std::string tor_proxy;
   if (options.GetString("tor_path", &tor_path) &&
-      options.GetString("tor_proxy", &tor_proxy)) {
+      options.GetString("tor_proxy", &tor_proxy_)) {
     url::Parsed url;
     url::ParseStandardURL(
```

Comment:
This has already parsed `scheme`, `host`, and `port`. Try storing this and pass them to `ProxyConfigServiceTor` and doing some changes in it

Reply:
I'd really like to pull all this tor-specific code of out BraveBrowserContext and pass in some kind of partitionable ProxyServiceFactory.

Reply:
there's really no reason that BraveBrowserContext needs to know anything at all about tor specifically